File: fixtures/python/empty_block/negative.py
# Fixture: negative cases for MAINT013-empty-block (Python)
# None of the following should produce a finding.

from abc import abstractmethod
from typing import overload, Protocol

x = 1

# Non-empty if
if x:
    print(x)

# Non-empty for
for i in range(10):
    print(i)

# Non-empty while
while False:
    print("never")


class Base:
    @abstractmethod
    def method(self) -> None:
        pass  # abstractmethod stub — skip


class Foo:
    @overload
    def bar(self, x: int) -> int:
        ...  # overload stub — skip


class MyProto(Protocol):
    def proto_method(self) -> None:
        ...  # Protocol stub — skip

File: fixtures/js/empty_block/negative.ts
// Fixture: negative cases for MAINT013-empty-block (JS/TS)
// None of the following should produce a finding.

const x = 1;

// Non-empty if
if (x) {
    console.log(x);
}

// Non-empty while
let count = 0;
while (count < 10) {
    count++;
}

// Non-empty for
for (let i = 0; i < 10; i++) {
    console.log(i);
}

// Intentional catch swallow with `_`
function safeCall() {
    try {
        doSomething();
    } catch (_) {}
}

declare function doSomething(): void;
